src/code/z_map_disp.c

```diff
@@ -1,109 +1,1881 @@
+        }
+    }
+    sMapDisp.mapDataScene = &sMapDataScene;
+    MapDisp_InitSceneFloorData(play);
+    MapDisp_InitBossRoomStorey(play);
+}
+
+/**
+ * @brief Creates a deep copy of chest data from the scene data.
+ *
+ * @param play
+ * @param num
+ * @param segmentAddress
+ * @note If a boss scene is loaded, no data is copied. This allows the scene to borrow the main dungeon scene data
+ * instead.
+ */
+void MapDisp_InitChestData(PlayState* play, s32 num, void* segmentAddress) {
+    MapDataChest* mapDataChests;
+    s32 i;
+
+    if (!Map_IsInBossScene(play)) {
+        mapDataChests = Lib_SegmentedToVirtual(segmentAddress);
+        for (i = 0; i < num; mapDataChests++, i++) {
+            sMapDataChests[i] = *mapDataChests;
+        }
+        sNumChests = num;
+    }
+    sMapDisp.mapDataChests = sMapDataChests;
+    sMapDisp.numChests = sNumChests;
+}
+
+/**
+ * @brief Creates a deep copy of transition actors from the scene data.
+ *
+ * @param play
+ * @param num number of transition actors within the list
+ * @param transitionActorList pointer to the list of transition actors
+ * @note If a boss scene is loaded, no data is copied. This allows the scene to borrow the main dungeon scene data
+ * instead.
+ */
+void MapDisp_InitTransitionActorData(PlayState* play, s32 num, TransitionActorEntry* transitionActorList) {
+    s32 i;
+
+    if (!Map_IsInBossScene(play)) {
+        sTransitionActorList.count = num;
+        for (i = 0; i < num; i++) {
+            sTransitionActors[i] = transitionActorList[i];
+        }
+        sTransitionActorList.list = sTransitionActors;
+    }
+}
+
+void MapDisp_Destroy(PlayState* play) {
+    s32 i;
+
+    sMapDisp.mapDataScene = NULL;
+    sMapDisp.curRoom = -1;
+    sMapDisp.minimapCurX = 210;
+    sMapDisp.minimapCurY = 140;
+    sMapDisp.minimapCurTex = NULL;
+    sMapDisp.prevRoom = -1;
+    sMapDisp.minimapPrevTex = NULL;
+    sMapDisp.minimapPrevX = 0;
+    sMapDisp.minimapPrevY = 0;
+    sMapDisp.unk20 = 0;
+    sMapDisp.swapAnimTimer = 0;
+    sMapDisp.texBuff0 = NULL;
+    sMapDisp.texBuff1 = NULL;
+
+    for (i = 0; i < sSceneNumRooms; i++) {
+        MapDisp_DestroyRoomStoreyRecord(play, &sMapDisp.roomStoreyList[i]);
+    }
+
+    sMapDisp.roomStoreyList = NULL;
+    sMapDisp.numStoreys = 0;
+    sMapDisp.pauseMapCurStorey = 0;
+    sMapDisp.bottomStorey = 0;
+    sMapDisp.timer = 0;
+    sMapDisp.storeyYList = NULL;
+    sMapDisp.numChests = 0;
+    sMapDisp.mapDataChests = NULL;
+    MapDisp_DestroyMapI(play);
+    sMapDisp.unk5A = 0;
+}
+
+void MapDisp_Update(PlayState* play) {
+    s16 currentX;
+    s16 currentY;
+    s16 targetX;
+    s16 targetY;
+
+    if ((sMapDisp.mapDataScene != NULL) && (sSceneNumRooms != 0)) {
+        //! FAKE: D_801BEBFA should be sMapDisp.pauseMapCurStorey
+        D_801BEBFA = DUNGEON_FLOOR_INDEX_0 - play->pauseCtx.cursorMapDungeonItem;
+        if (sMapDisp.prevRoom != -1) {
```

Comment:
Oh, I see why the extern is there.
I prefer the whole function to be marked as NON_MATCHING instead of introducing a fake symbol because this would break shiftability

Reply:
```suggestion
    if ((sMapDisp.mapDataScene != NULL) && (sSceneNumRooms != 0)) {
        //! FAKE: D_801BEBFA should be sMapDisp.pauseMapCurStorey
        sMapDisp.pauseMapCurStorey = DUNGEON_FLOOR_INDEX_0 - pauseCtx->cursorMapDungeonItem;
        if (sMapDisp.prevRoom != -1) {
```
You don't need the fake symbol if you create a `pauseCtx` stack variable and do